hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/QueueManager.java

```diff
@@ -272,6 +272,16 @@ private FSQueue createNewQueues(FSQueueType queueType,
       FSParentQueue newParent = null;
       String queueName = i.next();
 
+      // Check if child policy is allowed
```

Comment:
Yes,  my original thought is to do that in another JIRA. The depth and parent-child policy are not the same. It mighty a good idea to combine them since the logic of depth checking only prevent fifo policy to be non-leaf queue. The current implementation seems a bit heavy. I can do it in this JIRA.